basis/BasisFunction.py

```diff
@@ -0,0 +1,110 @@
+from abc import ABC, abstractmethod
```

Comment:
For modules with short names, prefer to just import to avoid polluting the namespace:
```suggestion
import abc
```